Gems/Meshlets/Code/Source/Meshlets/MeshletsData.h

```diff
@@ -0,0 +1,181 @@
+#include <AzCore/std/containers/map.h>
+#include <AzCore/std/containers/list.h>
+
+#include "../../External/meshoptimizer.h"
+
+namespace AZ
+{
+    namespace RPI
+    {
+        class MeshDrawPacket;
+    }
+
+    namespace Meshlets
+    {
+        enum class ComputeStreamsSemantics : uint8_t
+        {
+            MeshletsData = 0,
+            MehsletsTriangles,
+            MeshletsIndicesIndirection,
+
+            // for debug coloring purposes
+            UVs,
+            Indices,
+
+            NumBufferStreams
+        };
+
+        enum class RenderStreamsSemantics : uint8_t
+        {
+            Positions = 0,
+            Normals,
+            Tangents,
+            BiTangents,
+
+            UVs,
+            Indices,
+
+            NumBufferStreams
+        };
+
+        struct MeshletDescriptor
+        {
+            //! Offset into the indirect indices array representing the global index of
+            //! all the meshlet vertices.
+            //! The Indirect vertices array is built as follows:
+            //!     std::vector<uint32_t> indirectIndices;
+            //!     indirectIndices = { { meshlet 1 vertex indices }, { meshlet 2 }, .. { meshlet n} }
+            uint32_t vertexOffset;      // In uint32_t steps
+
+            //! Offset into the global meshlets triangleIndices array represented as:
+            //!     std::vector<uint8_t> triangleIndices;
+            //!     triangleIndices = { {meshlet 1 local indices group}, ... { meshlet n} }
+            //! The local indices are an 8 bits index that can represent up to 256 entries.
+            uint32_t triangleOffset;    // In bytes from the start of the array
+
+            //! Finding a vertex within the meshlet is done like that:
+            //!     triangleOffset = currentMeshlet.triangleOffset + meshletTrIndex * 3;
+            //!     localIndex_i = meshletTriangles[triangleOffset + i];    // i = triangle vertex index 0..2
+            //!     vertexIndex_i =  indirectIndices[currentMeshlet.vertexOffset + localIndex_i];
+
+            //! Amount of vertices and triangle for the mesh - based on this the arrays
+            //! indirectIndices and triangleIndices are created per meshlet.
+            uint32_t vertexCount;
+            uint32_t triangleCount;
+        };
+
+        struct GeneratorVertex
+        {
+            float px, py, pz;
+            float nx, ny, nz;
+            float tx, ty;
+        };
+
+        struct GeneratorMesh
+        {
+            std::vector<GeneratorVertex> vertices;
+            std::vector<unsigned int> indices;
+        };
+
+        //======================================================================
+        struct MeshletsData
+        {
+            std::vector<meshopt_Meshlet> Descriptors;
+            std::vector<uint32_t> EncodedTriangles;     // Meshlet triangles local indices [0..256]
+            std::vector<uint32_t> IndicesIndirection;   // Vertex Index indirection map - local to global
```

Comment:
Unfortunatelly this cannot be assumed since this is a connection point with external library and one should not assume raw data support between the two, especially when this is not guaranteed in the future.
The only way around this is to copy item by item and currently this does not seem relevant and definitely slower.